website/src/blogs/2018-10-02-setting-up-persistent-volumes-in-rwx-mode-using-openebs.md

```diff
@@ -0,0 +1,280 @@
+---
+title: Setting up persistent volumes in RWX mode using OpenEBS
+slug: setting-up-persistent-volumes-in-rwx-mode-using-openebs
+author: Ashish Ranjan
+date: 02-10-2018
+tags: Openebs, Solutions, wordpress, Nfs, Cloud Native Storage
+excerpt: Many stateful applications like WordPress require persistent storage in Read-Write-Many or RWX mode. OpenEBS is popular in the open-source community for its ease of use, and its simplistic design for pluggable storage engines.
+---
+
+Many stateful applications like WordPress require persistent storage in Read-Write-Many or RWX mode. OpenEBS is popular in the open-source community for its ease of use, and its simplistic design for pluggable storage engines.
+
+Currently, block volume or iSCSI support is provided natively by OpenEBS. Here, native iSCSI support means that the iSCSI stack is part of OpenEBS project and has full support for synchronous replication, granular monitoring, day 2 storage operations like backup restore, etc. iSCSI is typically used in Read-Write-Once or RWO mode which is common for all block storage engines. There have been multiple inquiries from the OpenEBS community about RWX support from OpenEBS, in other words, support for NFS. Though native support for NFS is being considered, it is already possible today to provision RWX mode storage using OpenEBS and to use it for applications like WordPress that require RXW mode.
+![](/content/images/2020/01/wordpress-with-openebs-over-nfs.png)
+In this blog, an example of WordPress is taken to show how OpenEBS storage volumes are exposed in RWX mode through the use of NFS in between WordPress and the Jiva volumes of OpenEBS.
+
+WordPress, when deployed on Kubernetes, requires both shared storage volumes on NFS and block storage volumes on iSCSI. The shared storage is required to store the core WordPress content or the admin managed content so that all the WordPress PODs can share the same data quickly. When PODs are spawned by Kubernetes on the fly to service more traffic, the PODs need to initialize quickly and require the core data to be available close to the application and in RWX mode. As an extremely high percentage of the shared storage traffic is Read traffic, the RWX volume need not be highly performant for writes and therefore the data can be served through NFS sitting in front of an iSCSI volume.
+
+The typical deployment of a scalable WordPress application is shown in the below diagram.
+![](/content/images/2020/01/wordpress-deployment-architecture.png)
```

Comment:
For this type of `/content/images/...`, please provide image to `public/images/blog/`

Reply:
Please check for other blog also for the same.